### ISSUE TITLE
FreeBSD support

### ISSUE THREAD

**Issue by MTM828** [mindstorm38/portablemc#105]
I was just wondering, can we have FreeBSD support?
FreeBSD does have a Linux compatibility layer, so the launcher should work fine when using it.
It's not really the launcher's fault because it's Minecraft that doesn't support FreeBSD.
I've tried doing it but I get an error complaining about LWJGL not being there, apparently it's because LWJGL doesn't have FreeBSD binaries. Though I've seen people make unofficial ports for FreeBSD, perhaps the launcher can use that?

Also, in `cli.py` it'd be better if it checks whether or not `get_minecraft_dir()` returns `None`, that way you can check if the OS is supported or not and return an error if it isn't (I encountered an error in `pat.join` saying `get_minecraft_dir()` is `NoneType`).

I know this is a big request and probably nobody is running FreeBSD anyways, FreeBSD is quite an unpopular OS so supporting it isn't really worth the trouble.
But if you do consider supporting FreeBSD, I'm always here to help. I can test the launcher on my system if you want.

**Comment by MTM828** [comment]
After installing LWJGL via `pkg` (the FreeBSD package manager), I had to do the following to get it to work:
- Rename `/usr/local/lib/lwjgl/liblwjgl64.so` to `lwjgl64.so` (this would be `liblwjgl.so` to `lwjgl.so` if I were on a 32-bit system)
- Change `java.library.path to `/usr/local/lib:/usr/local/lib/lwjgl:/usr/local/share/java/classes/lwjgl`
- Add FreeBSD to `get_minecraft_dir()` (same as Linux path)
The launcher installs the Linux LWJGL, but doesn't recognize it so it just ignores it.
By the way, this is on Minecraft 1.12, I'll test it with 1.19 in a moment.

**Comment by MTM828** [comment]
Turns out FreeBSD support is a long and complicated process with no official way to do it yet, just because LWJGL3 can't run (or even be built on) FreeBSD, because of how OS-specific the build process is (IMO, they did that on purpose so they have more control over the project).

Porting LWJGL2 has been successful though, and there's a package for it:`pkg install lwjgl`, all that's needed is to update the library path to `/usr/local/lib:/usr/local/lib/lwjgl:/usr/local/share/java/classes/lwjgl` and 1.12 should run, 1.13 and over use LWJGL3.
> Rename /usr/local/lib/lwjgl/liblwjgl64.so to lwjgl64.so (this would be liblwjgl.so to lwjgl.so if I were on a 32-bit system)

In other words, mess around with the names if it doesn't find the library.

**Comment by mindstorm38** [comment]
Sad to see that LWJGL3 is not even compiling...

**Comment by MTM828** [comment]
Yeah...
